Yoast/YoastSEO.js
# Improve keyword highlighting in snippet.

## Summary

This PR can be summarized in the following changelog entry:

## Relevant technical choices:

* Instead of stripping out special characters in the keyword, use lodash escape to escape them. 
* Removed the sanitize string from the snippet preview and let stringToRegex handle this (where this was already being done). 
* Updated some specs 

## Test instructions

This PR can be tested by following these steps:

* Use `Yoast SEO 3.8` as keyword and make a meta with the text `Yoast SEO 3.8` in it. See that it isn't highlighted. 
* Change `Yoast SEO 3.8` to `Yoast SEO` and see the keyword is highlighted. 
* Switch to this branch and see `Yoast SEO 3.8` is highlighted as expected. 

* Keywords with characters like `&` or `*` can now be highlighted.

Fixes the first part of https://github.com/Yoast/wordpress-seo/issues/6042
